### ISSUE TITLE
Apply window-specific bindings when the title/class changes on Sway

### ISSUE THREAD

**Issue by the-blank-x** [rvaiya/keyd#525]
When I open a window on sway (well, technically [swayfx](https://github.com/WillPower3309/swayfx)), keyd-application-mapper does not apply keybinds with a class expression (at least for pinentry-gtk2 and Librewolf) until I refocus the window. I don't know if this applies to all wlroots-based compositors. I believe this is because the window is focused before the mapper gets the class/app id, as I can see the following events happen if I apply [show-wlroots-events.patch.gz](https://github.com/rvaiya/keyd/files/11917885/show-wlroots-events.patch.gz):
- A new window is opened
- The new window has set its title to pinentry-gtk2
- My terminal is unfocused
- pinentry-gtk2 is focused (this is when the mapper resets keyd and applies window-specific bindings)
- pinentry-gtk2 has set its app id/class to Pinentry-gtk-2

Here are the snippets for the output with the patch applied:

<details>
<summary>Opening pinentry-gtk2</summary>

```
obj: 0x4
event: 0
payload (hex): 050000ff
payload (rep): b'\x05\x00\x00\xff'

obj: 0xff000005 (pinentry-gtk2)
event: 0
payload (hex): 0e00000070696e656e7472792d67746b32000000
payload (rep): b'\x0e\x00\x00\x00pinentry-gtk2\x00\x00\x00'

obj: 0xff000003 (foot)
event: 4
payload (hex): 00000000
payload (rep): b'\x00\x00\x00\x00'

obj: 0xff000005 (pinentry-gtk2)
event: 4
payload (hex): 0400000002000000
payload (rep): b'\x04\x00\x00\x00\x02\x00\x00\x00'

obj: 0xff000005 (pinentry-gtk2)
event: 1
payload (hex): 0f00000050696e656e7472792d67746b2d320000
payload (rep): b'\x0f\x00\x00\x00Pinentry-gtk-2\x00\x00'

obj: 0xff000005 (pinentry-gtk2)
event: 5
payload (hex):
payload (rep): b''

obj: 0xff000003 (foot)
event: 5
payload (hex):
payload (rep): b''
```
</details>

<details>
<summary>Opening LibreWolf</summary>

```
obj: 0x4
event: 0
payload (hex): 050000ff
payload (rep): b'\x05\x00\x00\xff'

obj: 0xff000005 (LibreWolf)
event: 0
payload (hex): 0a0000004c69627265576f6c66000000
payload (rep): b'\n\x00\x00\x00LibreWolf\x00\x00\x00'

obj: 0xff000003 (foot)
event: 4
payload (hex): 00000000
payload (rep): b'\x00\x00\x00\x00'

obj: 0xff000005 (LibreWolf)
event: 4
payload (hex): 0400000002000000
payload (rep): b'\x04\x00\x00\x00\x02\x00\x00\x00'

obj: 0xff000005 (LibreWolf)
event: 1
payload (hex): 0a0000004c69627265576f6c66000000
payload (rep): b'\n\x00\x00\x00LibreWolf\x00\x00\x00'

obj: 0xff000005 (LibreWolf)
event: 5
payload (hex):
payload (rep): b''

obj: 0xff000003 (foot)
event: 5
payload (hex):
payload (rep): b''
```
</details>

**Comment by the-blank-x** [comment]
https://github.com/rvaiya/keyd/pull/545 has a proper fix for this—not yet merged for whatever reason, but I've been running it fine ever since

**Comment by voronind-com** [comment]
@rvaiya I know this is not the right place to ask, but what are the plans for Application Mapper? It's been in development for a while, and basic issues like this one are not really addressed. Was the development dropped? Thanks.

**Comment by rvaiya** [comment]
@voronind-com Sorry for the delay, this one seems to have slipped through my radar (there is a lot of issue and PR noise). This is indeed a high value fix and has been merged in eb6fe4c1. 

> Was the development dropped?

The current implementation was originally intended as a stop-gap measure and will ultimately be replaced.

**Comment by voronind-com** [comment]
> @voronind-com Sorry for the delay, this one seems to have slipped through my radar (there is a lot of issue and PR noise). This is indeed a high value fix and has been merged in [eb6fe4c](https://github.com/rvaiya/keyd/commit/eb6fe4c167fdd8c68f398658c9226319a50d5a52).
> 
> > Was the development dropped?
> 
> The current implementation was originally intended as a stop-gap measure and will ultimately be replaced.

Thank you very much for taking action right now, really appreciate it! <3
Absolutely love the per-application mappings, high hopes for the new implementation. Is there some kind of tracker or blog for the progress?
